metrics/web/docs/features/deep_links/01_deep_linking.md

```diff
@@ -399,3 +412,214 @@ class MetricsRouteInformationParser {
 Consider this class diagram that illustrates the required changes needed to parse the deep links from the URL:
 
 ![Parsing deep links diagram](http://www.plantuml.com/plantuml/proxy?cache=no&fmt=svg&src=https://raw.githubusercontent.com/Flank/flank-dashboard/master/metrics/web/docs/features/deep_links/diagrams/parsing_deep_links_class_diagram.puml)
+
+#### Applying Deep Links
+The following subsections describe the required changes needed to be able to restore the application state from the deep links and update the deep links in response to application events.
+
+##### PageParametersModel
+Once we're able to parse `query parameters`, we should have a way to represent them in the Metrics Web application. To do that, let's introduce a `PageParametersModel`. 
+
+The `PageParametersModel` is an interface and does not store any data, however, it creates a general interface for all implementers including the `toMap() : Map<String, dynamic` method. 
+
+The implementers of the `PageParametersModel` should include the parameters that are specific to a `page` they relate to, and implement the `PageParametersModel` interface.
+
+The following class diagram demonstrates the general approach for creating new `PageParametersModel`s on a `SomePageParametersModel` example:
+
+![PageParametersModel class diagram](http://www.plantuml.com/plantuml/proxy?cache=no&fmt=svg&src=https://raw.githubusercontent.com/Flank/flank-dashboard/deep_links_design_improvements/metrics/web/docs/features/deep_links/diagrams/page_parameters_model_class_diagram.puml)
+
+##### PageParametersFactory
+Once we've implemented the required `PageParametersModel`, we should create a `PageParametersFactory` that encapsulates the creation of specific `PageParametersModel` from the given `RouteConfiguration`.
+
+Consider the following code snippet that demonstrates the `PageParametersFactory.create` method:
+```dart
+/// Returns the [PageParametersModel] based on the given [configuration].
+PageParametersModel create(RouteConfiguration configuration) {
+  final parameters = configuration.parameters;
+  
+  switch(configuration.name) {
+    case RouteName.dashboard:
+      return DashboardPageParameters.fromMap(parameters);
+    case RouteName.projectGroups:
+      /// Other page parameters handling
+  }
+}
+```
+
+The `PageParametersFactory` should be injected into `NavigationNotifier`, since the `NavigationNotifier` holds all the navigation-related logic of the application and controls the deep linking of the application. Consider the [`NavigationNotifier`](#navigationnotifier) section describing this part of the application.
```

Comment:
:label:
```suggestion
The `PageParametersFactory` should be injected into `NavigationNotifier` since the `NavigationNotifier` holds all the navigation-related logic of the application and controls its deep linking. Consider the [`NavigationNotifier`](#navigationnotifier) section to know more details in notifier-related changes.
```